doc/source/howtos/proxysql.rst

```diff
@@ -749,3 +749,71 @@ The script will detect the change and mark the node as
   +--------------+---------------+------+--------+
   3 rows in set (0.00 sec)
 
+.. _pxc-maint-mode:
+
+Maintenance Mode
+================
+
+Usually, to take a node down for maintenance, you need to identify that node,
+update its status in ProxySQL to ``OFFLINE_SOFT``,
+wait for ProxySQL to divert traffic from this node,
+and then initiate the shutdown or perform maintenance tasks.
+
+|PXC| includes a special *maintenance mode* for nodes
+that enables you to take a node down without adjusting ProxySQL manually.
+The mode is controlled using the :variable:`pxc_maint_mode` variable,
+which is monitored by ProxySQL and can be set to one of the following values:
+
+* ``DISABLED``: This is the default state
+  that tells ProxySQL to route traffic to the node as usual.
+
+* ``SHUTDOWN``: This state is set automatically
+  when you initiate node shutdown.
+
+  You may need to shut down a node when upgrading the OS, adding resources,
+  changing hardware parts, relocating the server, etc.
+
+  When you initiate node shutdown, |PXC| does not send the signal immediately.
+  Intead, it changes the state to ``pxc_maint_mode=SHUTDOWN``
+  and waits for a predefined period (60 seconds by default),
+  until existing transactions finish.
+  When ProxySQL detects that the mode is set to ``SHUTDOWN``,
+  it changes the status of this node to ``OFFLINE_SOFT``,
+  which stops creation of new connections for the node.
+  After the transition period,
+  any long-running transactions that are still active are aborted.
+
+* ``MAINTENANCE``: You can change to this state
+  if you need to perform maintenace on a node without shutting it down.
+
+  You may need to isolate the node for some time,
+  so that it does not receive traffic from ProxySQL
+  while you resize the buffer pool, truncate the undo log,
+  defragment or check disks, etc.
+
+  To do this, manually set ``pxc_maint_mode=MAINTENANCE``.
+  Control is not returned to the user for a predefined period
+  (60 seconds by default), until existing transactions finish.
+  When ProxySQL detects that the mode is set to ``SHUTDOWN``,
+  it stops routing traffic to the node.
+  Once control is returned,
+  any long-running transactions that are still active are aborted,
+  and you can perform maintenance activity.
+
```

Comment:
This doesn't apply to MAINTENANCE mode.
Existing large transaction will continue. ProxySQL will not start new transaction.
(In shutdown case server was going down)